bloom-housing/bloom
# feat: single use code request endpoint

## Issue Overview

This PR addresses [#issue](https://github.com/bloom-housing/bloom/issues/3899)


## Description
This creates a new endpoint that allows a user to request a single use code

## How Can This Be Tested/Reviewed?
Tests were added, but the easiest way to test this is a bit complex. Follow these steps:

- Update the `jurisdictions` table to set `allow_single_use_code_login` to be true
- create a user via the public site
- modify the public site's sign in code to also call the new `authService.requestSingleUseCode` in addition to logging in 
- go to log in (you'll probably get an error if you didn't confirm your user). You should have received an email that matches the mock https://www.figma.com/file/2LSD4B4yBIxAUGbZQCVOCv?node-id=2901:7177&mode=design#716447984

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
